### DISCUSSION TITLE
Standardising "getters"

### DISCUSSION THREAD

**Discussion by GerardoFurtado** [d3/d3#3547] [Q&A]
We are all quite used to the D3 way of passing no arguments to a method for getting its current value. For instance, `scale.domain()` will get the scale's domain, and so on...

However, this approach will fail for some methods: for instance, if I have a given arc generator, using `arc.outerRadius()` will not return me the current outer radius, but the accessor function instead, so I have to do `arc.outerRadius()()`.

What about standardising that approach? For the example above, the source code could be:

    arc.outerRadius = function(_) {
        return arguments.length ? (outerRadius = typeof _ === "function" ? _ : constant(+_), arc) : outerRadius();
        //calling outerRadius here -----------------------------------------------------------------------------^
    };

**Comment by Fil** [comment]
The outerRadius can be [specified](https://github.com/d3/d3-shape#arc_outerRadius) either as a constant or as a function of the datum, and the code normalizes constant values to a function. If we wanted to change that, then the API would return inconsistently either a function or a constant—and this would put the burden on any client code to check whether it's a function or a number, possibly retransforming the constant to a function. (Also, it would be backwards-compatible.)
